### PR TITLE
feat(shared-docs): create tutorial/playground generation

### PR DESCRIPTION
**Pull request by josephperrott**
Generate the tutorial/playground files

### CODE REVIEW

File: docs/markdown/tutorial/playground.ts [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+import {dirname, join} from 'path';
+import {FileAndContentRecord} from '../../interfaces';
+import {existsSync, mkdirSync, writeFileSync, readFileSync} from 'fs';
+import {addDirectoryToFilesRecord, findAllConfigs} from './utils';
+import {generateMetadata} from './metadata';
+import {generateSourceCode} from './source-code';
+import {generatePlaygroundRoutes} from './routes';
+import {glob} from 'glob';
+
+async function generatePlaygroundFiles(
```
Comment (jelbourn) [inline docs/markdown/tutorial/playground.ts:18]: Might be good to have a comment that explains what's going on there

File: docs/markdown/tutorial/tutorial.ts [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+
+import {dirname, basename, join} from 'path';
+import {FileAndContentRecord} from '../../interfaces';
+import {existsSync, mkdirSync, readFileSync, writeFileSync} from 'fs';
+import {addDirectoryToFilesRecord, findAllConfigs, findConfig} from './utils';
+import {generateMetadata} from './metadata';
+import {generateSourceCode} from './source-code';
+import {generateTutorialRoutes} from './routes';
+
+async function generateTutorialFiles(tutorialDir: string, commonDir: string, outputDir: string) {
```
Comment (jelbourn) [inline docs/markdown/tutorial/tutorial.ts:17]: Similar here, a comment that explains the idea of what's happen would be good

**Review by jelbourn — APPROVED** [review]
LGTM

**Comment by josephperrott** [comment]
This PR was merged into the repository by commit 6dc608da081ae606eee123481b2008d9dead50fe.

**Pull request closed without merging**